lib/cli/icgc-download-client.py

```diff
@@ -0,0 +1,74 @@
+import argparse
+import logging
+import yaml
+
+from clients.ega import ega_script
+from clients.gdc import gdc_script
+from clients.gnos import genetorrent
+from clients.icgc import icgc_script
+
+
+def config_parse(filename):
+    try:
+        configtext = open(filename, 'r')
+    except IOError:
+        print "Config file not found"
+    try:
+        config_temp = yaml.load(configtext)
+    except yaml.YAMLError:
+        print "Could not read config file"
+    return config_temp
+
+
+def logger_setup(logfile):
+    logger = logging.getLogger('__log__')
+    logger.setLevel(logging.DEBUG)
+    formatter = logging.Formatter('%(asctime)s - %(levelname)s - %(message)s')
+
+    fh = logging.FileHandler(logfile)
+    fh.setLevel(logging.DEBUG)
+    fh.setFormatter(formatter)
+    logger.addHandler(fh)
+
+    sh = logging.StreamHandler()
+    sh.setLevel(logging.INFO)
+    sh.setFormatter(formatter)
+    logger.addHandler(sh)
+    return logger
+
+
+
+parser = argparse.ArgumentParser()
+parser.add_argument("--config", nargs='?', default="/icgc/mnt/configuration/config.yaml",
+                    help="File used to set download preferences and authentication criteria")
+parser.add_argument('repo', choices=['ega', 'collab', 'cghub', 'aws', 'gdc'],
+                    help='specify which repository to download from, all lowercase letters')
+parser.add_argument('file_id', help='Lowercase identifier of file or path to manifest file')
```

Comment:
We should talk about the CLI. I'm not sure that `file_id` should be positional. I think an option is more appropriate as defined in the ticket.